modules/client-common/src/main/java/org/apache/ignite/client/proto/ClientMessagePacker.java

```diff
@@ -550,6 +573,75 @@ public ClientMessagePacker packObject(Object val) {
         throw new UnsupportedOperationException("Unsupported type, can't serialize: " + val.getClass());
     }
 
+    /**
+     * Packs an array of different objects.
+     *
+     * @param args Object array.
+     * @return This instance.
+     * @throws UnsupportedOperationException in case of unknown type.
+     */
+    public ClientMessagePacker packObjectArray(Object[] args) {
+        assert !closed : "Packer is closed";
+
+        if (args == null) {
+            packNil();
+
+            return this;
+        }
+
+        packArrayHeader(args.length);
+
+        for (Object arg : args) {
+            if (arg == null) {
+                packNil();
+
+                continue;
+            }
+
+            Class<?> cls = arg.getClass();
+
+            if (cls == Boolean.class)
+                packBoolean((Boolean)arg);
+            else if (cls == Byte.class) {
+                packInt(ClientDataType.INT8);
+                packByte((Byte)arg);
+            }
+            else if (cls == Short.class) {
+                packInt(ClientDataType.INT16);
+                packShort((Short)arg);
+            }
+            else if (cls == Integer.class) {
+                packInt(ClientDataType.INT32);
+                packInt((Integer)arg);
+            }
+            else if (cls == Long.class) {
+                packInt(ClientDataType.INT64);
+                packLong((Long)arg);
+            }
+            else if (cls == Float.class)
+                packFloat((Float)arg);
+            else if (cls == Double.class)
+                packDouble((Double)arg);
+            else if (cls == String.class)
+                packString((String)arg);
+            else if (cls == UUID.class) {
+                packInt(ClientDataType.UUID);
+                packUuid((UUID)arg);
+            }
+            else if (cls == byte[].class)
+                writeByteArray((byte[])arg);
+            else
+                throw new UnsupportedOperationException("Custom objects are not supported");
```

Comment:
Can this be replaced with packObject() ?

Reply:
I think it make sense to support all primitive arrays, maybe in a separate ticket.

Reply:
Unfortunately, packObject method cannot be used. We know nothing about agrs list, so we can't unpack the object in "unpackObject" method because we don't have the corresponding object type.